code/game/objects/structures.dm

```diff
@@ -112,7 +113,12 @@
 	if(!can_climb(user))
 		return
 
-	user.visible_message("<span class='warning'>[user] starts [flags_atom & ON_BORDER ? "leaping over":"climbing onto"] \the [src]!</span>")
+	var/intotrench = FALSE ///temporary var so that this proc doesnt have to run istype twice
+	if(istype(src, /obj/structure/trench))
+		user.visible_message("<span class='warning'>[user] starts to climb into \the [src]!</span>")
+		intotrench = TRUE
```

Comment:
Dnt use istype(src, typepath)
You should be using subtypes here

Reply:
Wait what? 
How?

Reply:
Should be done.